### ISSUE TITLE
Option to test connectors without deploying them

### ISSUE THREAD

**Issue by riferrei** [gunnarmorling/kcctl#13]
Apache Pulsar also has a CLI that allows developers to handle connectors with ease. There is an option, though, that also allows developers to test connectors without fully deploying them. It is called [locarun](https://pulsar.apache.org/docs/en/io-cli/#localrun).

An example of usage of this functionality can be seen in the following video:

https://youtu.be/60x5eMeBEyI?t=1399

It would be very nice if the `kcctl` could provide similar functionality. Currently, one of the main sources of struggle with Kafka users is having to have a complete stack up-and-running to then — be able to test things out. Things like Docker and TestContainers help, but there is still a lot of plumbing before developers start playing with connectors.

**Comment by gunnarmorling** [comment]
Thanks for logging this, a very interesting idea. Some things we'll need to think about:

* This likely would only work in JVM mode, as additional code needs to be loaded dynamically (perhaps _kcctl_ might spin up a separate JVM with the connector)
* For source connectors, records could be simply printed out on the CLI; how would it work for sink connectors, would the user have to specify records, as with the console producer?

**Comment by riferrei** [comment]
> This likely would only work in JVM mode, as additional code needs to be loaded dynamically (perhaps kcctl might spin up a separate JVM with the connector)

Yes, this is the idea. Pulsar does this by spinning up a thread that holds the context of the execution of the connector. Any dependencies would have to be loaded into that same classloader.

> For source connectors, records could be printed out on the CLI; how would it work for sink connectors, would the user have to specify records, as with the console producer?

I think the best (and easy) way would be just focusing on the connector one-off deployment. For sources, checking if it is working would be just a matter of seeing if the records are being written into the Kafka topics, whereas for sinks, checking if it is working would be writing off records to existing topics and seeing if they are being pulled from the connector to be sent to the target system. This is how `localrun` from Pulsar works, and the developer experience is incredible
